packages/idyll-document/src/index.js

```diff
@@ -72,7 +72,7 @@ class IdyllDocument extends React.Component {
     this.setState({ error: error.message });
   }
 
-  componentWillReceiveProps(newProps) {
```

Comment:
Again, we can probably refactor this to use componentDidUpdate as well. This one might be the most work since there's a fair amount of logic in here

Reply:
Ok yeah I need to step through a few times and then can take a swing at it.

Reply:
I left this one off latest commit, going to find more idyll projects / build something to get more acquainted

Reply:
Yeah exactly, was thinking to separate.